pkg/controller/ec2/vpcpeeringconnection/setup.go

```diff
@@ -0,0 +1,189 @@
+package vpcpeeringconnection
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/ec2"
+	svcsdkapi "github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/ec2/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+// SetupVPCPeeringConnection adds a controller that reconciles VPCPeeringConnection.
+func SetupVPCPeeringConnection(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(svcapitypes.VPCPeeringConnectionGroupKind)
+	opts := []option{
+		func(e *external) {
+			c := &custom{client: e.client, kube: e.kube}
+			e.postObserve = c.postObserve
+			e.preObserve = preObserve
+			e.preDelete = c.preDelete
+			e.postCreate = c.postCreate
+			e.preCreate = preCreate
+			e.isUpToDate = c.isUpToDate
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&svcapitypes.VPCPeeringConnection{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.VPCPeeringConnectionGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type custom struct {
+	kube   client.Client
+	client svcsdkapi.EC2API
+}
+
+func preObserve(ctx context.Context, cr *svcapitypes.VPCPeeringConnection, obj *svcsdk.DescribeVpcPeeringConnectionsInput) error {
```

Comment:
I think overriding `filterList` would be more appropriate for this task.

Reply:
@muvaf Can you provide an example or further description?

Reply:
Take a look at https://github.com/crossplane/provider-aws/blob/959cc5b741fd7c2dafa7e20cac75ab55ae2c79bb/pkg/controller/rds/dbcluster/setup.go#L123